providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/grpc/EventStreamObserver.java

```diff
@@ -107,16 +112,13 @@ private void handleConfigurationChangeEvent(EventStreamResponse value) {
         this.onConnectionEvent.accept(true, changedFlags);
     }
 
+    /**
+     * Handles provider readiness events by clearing the cache (if enabled) and notifying listeners of readiness.
+     */
     private void handleProviderReadyEvent() {
         this.onConnectionEvent.accept(true, Collections.emptyList());
-        if (this.cache.getEnabled()) {
+        if (this.cache.getEnabled().equals(Boolean.TRUE)) {
```

Comment:
[question] Do we always want to purge the cache on a reconnect, that means even from Stale to ready? - (sidenote: need to check what is in the gherkin file for this)

Reply:
i don't know, this change was only suggested by sonar.

Reply:
I think we do need to purge here ; we could have missed change events while we were stale, so now that we are connected we should build up a new cache.